src/MessageHandler/HashFileHandler.php

```diff
@@ -2,14 +2,75 @@
 
 namespace App\MessageHandler;
 
-use App\Message\HashFile;
-
+use Doctrine\ORM\EntityManagerInterface;
+use League\Flysystem\FilesystemInterface;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
 
+use App\Message\HashFile;
+use App\Entity\File;
+use App\Repository\FileRepository;
+
+/**
+ * The handler for file hashing messages
+ */
 class HashFileHandler implements MessageHandlerInterface
 {
-    public function __invoke(HashFile $HashFile)
+    /**
+     * The entity handler.
+     *
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * The file Repository.
+     *
+     * @var FileRepository
+     */
+    private $fileRepository;
+
+    /**
+     * The monolog logger.
+     *
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Constructor for this Controller, to set up default services.
+     *
+     * @param EntityHandler   $entityHandler           The entity handler.
+     * @param FileRepository  $fileRepository          The file Repository.
+     * @param LoggerInterface $datasetFileHasherLogger Name hinted dataset_file_hasher logger.
+     */
+    public function __construct(EntityManagerInterface $entityManager, FileRepository $fileRepository, LoggerInterface $datasetFileHasherLogger)
+    {
+        $this->entityManager = $entityManager;
+        $this->fileRepository = $fileRepository;
+        $this->logger = $datasetFileHasherLogger;
```

Comment:
I think this needs a bind in the servics.yaml 
`        bind:
            Symfony\Bridge\Monolog\Logger $logger: '@monolog.logger.dataset_file_hasher'`

Reply:
No, this is a new shortcut you can use. see: https://symfony.com/doc/current/logging/channels_handlers.html#how-to-autowire-logger-channels

Reply:
That's pretty cool 👍